src/main/java/com/amazonaws/encryptionsdk/kms/AwsKmsMrkAwareMasterKeyProvider.java

```diff
@@ -193,6 +193,25 @@ public AwsKmsMrkAwareMasterKeyProvider.Builder withClientBuilder(AWSKMSClientBui
      */
     public AwsKmsMrkAwareMasterKeyProvider buildDiscovery() {
       final boolean isDiscovery = true;
+      if (discoveryMrkRegion_ == null) {
+        // The AWS SDK has a default process for evaluating the default Region. This returns null if
+        // no
+        // default region is found. Because a default region _may_ not be needed.
+        // Note: Final Null check is at
+        // com.amazonaws.encryptionsdk.kms.AwsKmsMrkAwareMasterKeyProvider#AwsKmsMrkAwareMasterKeyProvider#L417
+        try {
+          // = compliance/framework/aws-kms/aws-kms-mrk-aware-master-key-provider.txt#2.6
+          // # In
+          // # discovery mode if a default MRK Region is not configured the AWS SDK
+          // # Default Region MUST be used.
+          discoveryMrkRegion_ =
+              new com.amazonaws.regions.DefaultAwsRegionProviderChain().getRegion();
```

Comment:
Since we're directly calling Default Provider Chain, we could remove `getSdkDefaultRegion`
```
private static String getSdkDefaultRegion() {
      try {
        return new com.amazonaws.regions.DefaultAwsRegionProviderChain().getRegion();
      } catch (SdkClientException ex) {
        return null;
      }
    }
```